### PR TITLE
CI: Migrate Pull request workflow - Part1

### PR DESCRIPTION
**Pull request by phiz71**
## Issue

https://gravitee.atlassian.net/browse/APIM-2635

## Description

A first PR to add common jobs, and refactor a little bit existing jobs

### CODE REVIEW

File: .circleci/ci/src/jobs/job-deploy-on-azure.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { orbs } from '../orbs';
+import { CircleCIEnvironment } from '../pipelines';
+import { isMasterBranch } from '../utils';
+import { DockerExecutor } from '@circleci/circleci-config-sdk/dist/src/lib/Components/Executors';
+import { AzureCliExecutor } from '../executors/executor-azure-cli';
+
+export class DeployOnAzureJob {
+  private static jobName = 'job-deploy-on-azure-cluster';
+  public static create(dynamicConfig: Config, environment: CircleCIEnvironment): Job {
+    const k8sNamespace = `apim-apim-${environment.branch.replace('.', '-')}`;
```
Comment (jourdiw) [inline .circleci/ci/src/jobs/job-deploy-on-azure.ts:13]: Double apim?

Comment (phiz71) [reply .circleci/ci/src/jobs/job-deploy-on-azure.ts:13]: yes

File: .circleci/ci/src/jobs/job-perf-lint-build.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { commands, Config, Job, reusable } from '@circleci/circleci-config-sdk';
+import { Command } from '@circleci/circleci-config-sdk/dist/src/lib/Components/Commands/exports/Command';
+import { NodeLtsExecutor } from '../executors';
+import { NotifyOnFailureCommand, WebuiInstallCommand } from '../commands';
+
+export class PerfLintBuildJob {
```
Comment (jourdiw) [inline .circleci/ci/src/jobs/job-perf-lint-build.ts:6]: Should this be in the "frontend" folder as well?

**Review by jourdiw — APPROVED** [review]
LGTM. Comment non-blocking.

File: .circleci/ci/src/jobs/job-perf-lint-build.ts (hunk 2 above)
Comment (phiz71) [reply .circleci/ci/src/jobs/job-perf-lint-build.ts:6]: no, perf is a independant module.
But I think in the future, I will rename frontend in "node" or "npm". Don't know yet

Comment (jourdiw) [reply .circleci/ci/src/jobs/job-perf-lint-build.ts:6]: So "frontend" is more so for a specific workflow? As far as I could tell, `job-perf-lint-build` is almost identical to `job-webui-lint-test` except for saving the artifacts at the end.

Comment (phiz71) [reply .circleci/ci/src/jobs/job-perf-lint-build.ts:6]: and one of the run command which "builds" instead of "tests"

**Review by jourdiw — APPROVED** [review]

**Pull request merged**